server/apps/microservices/src/processors/machine-learning.processor.ts

```diff
@@ -45,24 +45,47 @@ export class MachineLearningProcessor {
     try {
       const { asset }: { asset: AssetEntity } = job.data;
 
-      const res = await axios.post(
-        immich_machine_learning_url + '/object-detection/detect-object',
-        {
-          thumbnailPath: asset.resizePath,
-        },
-      );
+      const res = await axios.post(immich_machine_learning_url + '/object-detection/detect-object', {
+        thumbnailPath: asset.resizePath,
+      });
 
       if (res.status == 201 && res.data.length > 0) {
-        const smartInfo = new SmartInfoEntity();
-        smartInfo.assetId = asset.id;
-        smartInfo.objects = [...res.data];
+        this.createAndSaveTagToAsset(asset, res.data);
+      }
+    } catch (e) {
+      this.logger.error(e);
+    }
+  }
+
+  private async createAndSaveTagToAsset(asset: AssetEntity, tagNames: string[]) {
+    try {
+      const immichTags: TagEntity[] = [];
+
+      for (const tag of tagNames) {
+        let immichTag: TagEntity | null;
+
+        immichTag = await this.tagRepository.findOne({ where: { name: tag, userId: asset.userId } });
```

Comment:
This may result in tagging assets with an existing tag that is not type OBJECT (eg, a tag that the user created manually). Is that a problem? I think the alternative is to log and skip if a non-object tag with the name already exists.

Reply:
I think it makes sense to keep these AI tags regardless of what custom tags have been added, but the UI can show a consolidated list if you are worried about showing duplicates.

Reply:
I see there is a UQ for name/userId, so maybe the design is to have unique tag names across types (sources), in which case it seems like it should be skipped if it already exists, since you'd either (1) get a unique key constrain violation or (2) risk updating the type from CUSTOM to OBJECT.